stdlib/public/core/Set.swift

```diff
@@ -1503,163 +1505,208 @@ internal protocol _SetBuffer { // FIXME: Remove or refactor for Set.
 @_fixed_layout // FIXME(sil-serialize-all)
 @usableFromInline
 @_objc_non_lazy_realization
-internal class _RawNativeSetStorage: _SwiftNativeNSSet, _NSSetCore {
-  @usableFromInline // FIXME(sil-serialize-all)
+internal class _SwiftRawSetStorage: _SwiftNativeNSSet {
+  // The _HashTable struct contains pointers into tail-allocated storage, so
+  // this is unsafe and needs `_fixLifetime` calls in the caller.
+  @usableFromInline
   @nonobjc
-  internal final var bucketCount: Int
+  internal final var hashTable: _HashTable
 
-  @usableFromInline // FIXME(sil-serialize-all)
-  internal final var count: Int
+  @usableFromInline
+  internal var seed: (UInt64, UInt64)
 
-  @usableFromInline // FIXME(sil-serialize-all)
-  internal final var initializedEntries: _UnsafeBitMap
+  @usableFromInline
+  @nonobjc
+  internal final var rawElements: UnsafeMutableRawPointer
 
-  @usableFromInline // FIXME(sil-serialize-all)
+  // This type is made with allocWithTailElems, so no init is ever called.
+  // But we still need to have an init to satisfy the compiler.
   @nonobjc
-  internal final var keys: UnsafeMutableRawPointer
+  internal init(_doNotCallMe: ()) {
+    _sanityCheckFailure("Only create this by using the `empty` singleton")
+  }
 
-  @usableFromInline // FIXME(sil-serialize-all)
-  internal final var seed: (UInt64, UInt64)
+  internal func invalidate() {}
+}
 
-  // This API is unsafe and needs a `_fixLifetime` in the caller.
-  @inlinable // FIXME(sil-serialize-all)
-  @nonobjc
-  internal final
-  var _initializedHashtableEntriesBitMapBuffer: UnsafeMutablePointer<UInt> {
-    return UnsafeMutablePointer(Builtin.projectTailElems(self, UInt.self))
+@usableFromInline
+internal class _SwiftEmptySetStorage: _SwiftRawSetStorage {
+  @objc
+  internal required init(objects: UnsafePointer<AnyObject?>, count: Int) {
+    _sanityCheckFailure("Don't call this designated initializer")
   }
+}
 
+extension _SwiftRawSetStorage {
   /// The empty singleton that is used for every single Dictionary that is
   /// created without any elements. The contents of the storage should never
   /// be mutated.
   @inlinable
   @nonobjc
-  internal static var empty: _RawNativeSetStorage {
+  internal static var empty: _SwiftEmptySetStorage {
     return Builtin.bridgeFromRawPointer(
       Builtin.addressof(&_swiftEmptySetStorage))
   }
+}
 
-  // This type is made with allocWithTailElems, so no init is ever called.
-  // But we still need to have an init to satisfy the compiler.
-  @nonobjc
-  internal init(_doNotCallMe: ()) {
-    _sanityCheckFailure("Only create this by using the `empty` singleton")
-  }
 
+extension _SwiftEmptySetStorage: _NSSetCore {
```

Comment:
FYI we did find it worthwhile in Foundation to have 1 element variants as well in addition to this strategy of a zero element container.

Couldn't this be a singleton?